public/app/plugins/datasource/loki/components/LokiQueryEditor.tsx

```diff
@@ -67,7 +68,8 @@ export const LokiQueryEditor = React.memo<LokiQueryEditorProps>((props) => {
     setDataIsStale(false);
   }, [data]);
 
-  const onChangeInternal = (query: LokiQuery) => {
+  const onChangeInternal = async (query: LokiQuery) => {
+    setQueryStats(await datasource.queryStatsRequest(query));
```

Comment:
I think we can move this to the `LokiQueryBuilderOptions` component, since `query` is already part of the properties. This would also take away the bad sideeffect, that `onChange(query);` is executed after the `/stats` call returned. Which potentially slows things down by a lot.